src/devices/Ads1115/Ads1115.cs

```diff
@@ -104,14 +220,255 @@ private void SetConfig()
+        {
+            if (gpioController == null)
+            {
+                throw new ArgumentNullException(nameof(gpioController));
+            }
+
+            if (queueLength == ComparatorQueue.Disable)
+            {
+                throw new ArgumentException("Must set the ComparatorQueue to something other than disable.", nameof(queueLength));
+            }
+
+            if (upperValue <= lowerValue)
+            {
+                throw new ArgumentException("Lower comparator limit must be larger than upper comparator limit");
+            }
+
+            if (_gpioController != null)
+            {
+                throw new InvalidOperationException("Event mode already set.");
+            }
+
+            try
+            {
+                // The ALRT/RDY Pin requires a pull-up resistor
+                gpioController.OpenPin(pin, PinMode.InputPullUp);
+
+                _comparatorQueue = queueLength;
+                _comparatorMode = mode;
+                // Event callback mode is only useful in Continuous mode
+                _deviceMode = DeviceMode.Continuous;
+                SetConfig();
+                // Writing a negative value to the max value register and a positive value to the min value register
+                // configures the ALRT/RDY pin to trigger after each conversion (with a transition from high to low, when ComparatorPolarity is Low)
+                WriteComparatorRegisters(lowerValue, upperValue);
+
+                gpioController.RegisterCallbackForPinValueChangedEvent(pin, ComparatorPolarity == ComparatorPolarity.Low ? PinEventTypes.Falling : PinEventTypes.Rising, ConversionReadyCallback);
+
+            }
+            catch (Exception)
+            {
+                gpioController.ClosePin(pin);
+                throw;
+            }
+
+            _gpioController = gpioController;
+            _alrtRdyPin = pin;
+        }
+
+        private ushort ReadConfigRegister()
+        {
+            Span<byte> retBuf = stackalloc byte[2];
+            Span<byte> request = stackalloc byte[1]
+            {
+                (byte)Register.ADC_CONFIG_REG_ADDR
+            };
+
+            _i2cDevice.WriteRead(request, retBuf);
+            return BinaryPrimitives.ReadUInt16BigEndian(retBuf);
+        }
+
+        /// <summary>
+        /// Wait until the current conversion finishes.
+        /// This method must only be called in powerdown mode, otherwise it would timeout, since the busy bit never changes.
+        /// Due to that, we always write the configuration first in power down mode and then enable the continuous bit.
+        /// </summary>
+        /// <exception cref="TimeoutException">A timeout occured waiting for the ADC to finish the conversion</exception>
+        private void WaitWhileBusy()
+        {
+            // In powerdown-mode, wait until the busy bit goes high
+            ushort reg = ReadConfigRegister();
+            int timeout = 5000;
```

Comment:
nit: I'd prefer this piece of code was written in such a way that units of timeout are clear so it's easier to adjust it in the future.

Reply:
Yea, this is something I should have seen myself.